src/Neuron/Zettelkasten/Tag.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Neuron.Zettelkasten.Tag
+  ( Tag (..),
+    TagPattern (..),
+    literalPattern,
+    tagPatternToText,
+    tagMatch,
+    isSubTag,
+    isStrictSubTag,
+  )
+where
+
+import Data.Aeson
+import Relude
+import Relude.Extra.Foldable1
+import System.FilePath
+import System.FilePattern as FilePattern
+
+newtype Tag = Tag {tagToText :: Text}
+  deriving (Eq, Ord, Show, ToJSON, FromJSON)
+
+instance Semigroup Tag where
+  Tag t <> Tag t' = Tag (toText $ toString t </> toString t')
+  sconcat tags = Tag (toText $ foldl1' (</>) $ fmap (toString . tagToText) tags)
+
+newtype TagPattern = TagPattern {toFilePattern :: FilePattern}
+  deriving (Eq, Show)
+
+instance Semigroup TagPattern where
+  TagPattern p <> TagPattern p' = TagPattern (p </> p')
```

Comment:
Use `GeneralizedNewtypeDeriving`?

Reply:
Oh wait, let's just use `</>` directly instead.

Reply:
Do you mean defining `type Tag = FilePath` directly?

Reply:
Let's take a step back and ask: where is it that you need to concatenate two tag patterns? If there is no need for it, let's eliminate this instance entirely.

Reply:
Right, this isn't needed anymore if I remove `isSubTag` and `isStrictSubTag` 👍